source/runtime.cpp

```diff
@@ -128,6 +128,28 @@ bool reshade::runtime::on_init(input::window_handle window)
 
 	return true;
 }
+
+bool reshade::runtime::on_init_vr()
```

Comment:
Think it would be easier to keep `on_init` and just add some special handling in there for when no window handle exists (handle is `nullptr`). See (which just sets an empty input object when there is no window):
https://github.com/crosire/reshade/blob/c1de0e14b68ba452fd3b3806f77c0eb503fffd90/source/runtime.cpp#L140
Also need to move the `assert` out of the `input` constructor in that case:
https://github.com/crosire/reshade/blob/c1de0e14b68ba452fd3b3806f77c0eb503fffd90/source/input.cpp#L55

Reply:
Agreed, that's much better. I initially didn't fully understand what implications this would have, so didn't want to touch existing code too much :)